src/plugins/creditor/credNodeStore.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import {type CreditorNode, type NodeUuid, type NodeTagId} from "./creditorNode";
+import {JsonLog} from "../../util/jsonLog";
+
+export type NodeEntry = {|type: "NODE", entry: CreditorNode|};
+export type NodeDetails = {title: string, id: NodeUuid};
+type NodesFile = JsonLog<NodeEntry>;
+
+export function toNodeEntry(entry: CreditorNode): NodeEntry {
+  return {type: "NODE", entry};
+}
+
+export function fromNodeEntry({entry}: NodeEntry): [NodeUuid, CreditorNode] {
+  return [entry.id, entry];
+}
+
+export function getDetails({id, title}: CreditorNode): NodeDetails {
+  return {id, title};
+}
+
+/**
+ * CredNodeStore is the static, in-memory representation of creditor nodes
+ *
+ * This class is used to accept and validate frontend inputs, and export them in
+ * serialized format utilizing the jsonLog functionality when saving to a file.
+ * Nodes in this class are still modifiable, of course, and jsonLogging is utilized
+ * because it outputs a gitdiff-friendly format
+ *
+ * Nodes in the the cred node store are not deletable at this time. In lieu of deleting nodes,
+ * their weights (`mint` values) can simply be set to zero
+ */
+export default class CredNodeStore {
+  +_nodes: Map<NodeUuid, CreditorNode>;
+
+  constructor(fromDisk: NodesFile | void) {
+    if (!fromDisk) {
+      this._nodes = new Map();
+    } else {
+      const nodeKVArray = Array.from(fromDisk.values()).map(fromNodeEntry);
+      this._nodes = new Map(nodeKVArray);
+    }
+  }
```

Comment:
Directly setting _nodes is a bad pattern. We will want to track more pieces of state (e.g. tag and parent relationships), and if we directly set _nodes, then we'll need to re-implement the state tracking here. That implies a class of bugs where we update the state inconsistently. It will be much safer to call _addNodes.

To that end, let's not overload the constructor where it can take a NodesFile. All this does is introduce the opportunity to code subtle bugs where the state is different depending on whether we loaded the nodes file directly in the constructor, or called the `setNode` API. Instead, let's have the constructor take no arguments, and have a static method which creates a CredNodeStore from the NodesFile. See Graph for example:

https://github.com/sourcecred/sourcecred/blob/8ade386780654e69a10b134d67d2b9ac66ac1c79/src/core/graph.js#L236-L251

https://github.com/sourcecred/sourcecred/blob/8ade386780654e69a10b134d67d2b9ac66ac1c79/src/core/graph.js#L836-L865